server/core_leaderboard.go

```diff
@@ -339,6 +340,26 @@ func LeaderboardRecordsList(ctx context.Context, logger *zap.Logger, db *sql.DB,
 		_ = rows.Close()
 	}
 
+	// Sort owner records according to leaderboard ordering
+	var sortFn func(i, j int) bool
+	if leaderboard.SortOrder == LeaderboardSortOrderAscending {
+		sortFn = func(i, j int) bool {
+			if ownerRecords[i].Score == ownerRecords[j].Score {
+				return ownerRecords[i].Subscore < ownerRecords[j].Subscore
+			}
+			return ownerRecords[i].Subscore < ownerRecords[j].Subscore
```

Comment:
Hm, should it be something like `return ownerRecords[i].Score < ownerRecords[j].Score` instead? :thinking:

Reply:
Good catch